### ISSUE TITLE
Iot endpoint is incorrect when using weksocket connect in China.

### ISSUE THREAD

**Issue by brandonbai** [aws-amplify/aws-sdk-android#582]
**Describe the bug**

When I use the endpoint like "xxxxx.ats.iot.cn-north-1.amazonaws.com.cn", the callback always output status = Reconnecting.

Then I found out the cause of this problem, just like I found this problem([Update AwsIotEndpointUtility.java #580](https://github.com/aws-amplify/aws-sdk-android/pull/580)).

step 1. 
```
com.amazonaws.mobileconnectors.iot.AwsIotEndpointUtility.java#getAccountPrefixFromEndpont
```
```java
    static String getAccountPrefixFromEndpont(String endpoint) {
// endpoint = "abc123.ats.iot.cn-north-1.amazonaws.com.cn"
        String endpointWithoutPort = stripPort(endpoint);
        validateIotEndpoint(endpointWithoutPort);
// return "abc123" instead of "abc123.ats".
        return splitEndpoint(endpointWithoutPort)[ENDPOINT_PREFIX_OFFSET];
    }
```
step 2. 
```
com.amazonaws.mobileconnectors.iot.AWSIotMqttManager.java#Line:728
```
```java
final String endpoint = String.format("%s.iot.%s.%s:443", accountEndpointPrefix, region.getName(),
                        region.getDomain());
// endpoint = "abc123.iot.cn-north-1.amazonaws.com.cn" instead of "abc123.ats.iot.cn-north-1.amazonaws.com.cn" 
```

So, Why not use endpoints directly, you just need to validate its format using regular expressions.

**Which AWS service(s) are affected?**
IoT

**Comment by minbi** [comment]
Hi @brandonbai ,

We have released a fix in version 2.8.4 of the SDK. Can you check whether this fixes the issue for you?

**Comment by brandonbai** [comment]
> Hi @brandonbai ,
> 
> We have released a fix in version 2.8.4 of the SDK. Can you check whether this fixes the issue for you?

The problem has been solved, thank you!

**Comment by minbi** [comment]
This is great to hear. Closing.
